csharp/src/Google.Protobuf.Test/LegacyGeneratedCodeTest.cs

```diff
@@ -144,15 +145,16 @@ public void LegacyGeneratedCodeThrowsWithIBufferWriter()
                 WriteContext.Initialize(new TestArrayBufferWriter<byte>(), out WriteContext writeCtx);
                 ((IBufferMessage)message).InternalWriteTo(ref writeCtx);
             });
-            Assert.AreEqual($"Message {typeof(LegacyGeneratedCodeMessageA).Name} doesn't provide the generated method that enables WriteContext-based serialization. You might need to regenerate the generated protobuf code.", exception.Message);
+            Assert.AreEqual($"Message {typeof(LegacyGeneratedCodeMessageA).Name} doesn't provide the generated method that enables WriteContext-based serialization. You might need to regenerate the generated protobuf code.", exception!.Message);
         }
 
         // hand-modified version of a generated message that only provides the legacy
         // MergeFrom(CodedInputStream) method and doesn't implement IBufferMessage.
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
         private sealed partial class LegacyGeneratedCodeMessageA : pb::IMessage {
-          private pb::UnknownFieldSet _unknownFields;
+            private pb::UnknownFieldSet _unknownFields;
```

Comment:
Let's keep the indentation consistent within this class - I can create a separate PR to reformat the whole of this bit, should we wish.